impl/pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,75 @@
+package telemetry
+
+import (
+	"context"
+	"time"
+
+	"github.com/TBD54566975/did-dht-method/config"
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/contrib/instrumentation/runtime"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const scopeName = "github.com/TBD54566975/did-dht-method"
+
+var (
+	Tracer        trace.Tracer
```

Comment:
hmm, don't love the idea of having a global tracer

can we put this on a struct which `SetupTelemetry` returns?

Reply:
Totally could. Is there a problem with having a global tracer? the gin middleware makes it's own tracer. An alternative to having a tracer that gets passed around everywhere is to initialize a tracer in each package that we can use to instrument that package.

Reply:
Global state scares me generally because of assumptions it caries - may/not be initialized, different callers using it for different purposes in an unsafe manner. Of course this is negligible given how small the codebase is now, though I view it as a best practice to avoid globally mutable state

Reply:
Reasonable. I'll figure out how I want to setup non-global tracer(s)

Reply:
So I refactored a bit to have a `GetTracer()` function that returns a tracer. There's still one instance that gets initialized on first call, which isnt great, but I'm not sure what better options there are. I could make a function + tracer instance like this for each package (then there wouldnt be a global one technically) but I dont see how that's any better. Let me know if you this this is acceptable, if not let's discuss options.

Reply:
I remember going through a similar type of conversation with @andresuribe87 on something a while back and he convinced me to rip out all the (complex, ugly) singleton code and just make a struct that returns a unique thing, which worked for the 90% case.

I believe what you've done here is fine too, and we should be open to changing it should that assumption be false

Reply:
I could see doing that too. It would involve passing it around through a lot of different functions, which seems annoying. I'm going to merge this as is, but i'm not opposed to moving to that in the future.